src/pages/BrowseBook/BrowseDocumentModel.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import {Link} from 'react-router-dom';
+
+const {ScriptureParaDocument} = require('proskomma-render');
+
+class BrowseDocumentModel extends ScriptureParaDocument {
+
+    constructor(result, context, config) {
+        super(result, context, config);
+        config.rendered = [];
+        config.blockStack = [];
+        config.currentChapter = null;
+        config.currentVerses = null;
+        addActions(this);
+    }
+}
+
+const addActions = (dInstance) => {
+    dInstance.addAction(
+        'startDocument',
+        () => true,
+        renderer => {
+            renderer.config.nextKey = 0;
+        }
+    );
+    dInstance.addAction(
+        'startBlock',
+        () => true,
+        renderer => {
+            renderer.config.blockStack = [];
+        }
+    );
+    dInstance.addAction(
+        'token',
+        () => true,
+        (renderer, context, data) => {
+            if (data.subType === 'wordLike') {
+                const attScopes = Array.from(context.sequenceStack[0].openScopes)
+                    .filter(s => s.startsWith('attribute'))
+                    .filter(s => ['strong', 'lemma', 'x-strong', 'x-lemma', 'x-content'].includes(s.split('/')[3]))
+                    .map(s => [s.split('/')[3], s.split('/')[5]]);
+                const state = {newSearchString: data.payload};
+                attScopes.forEach(s => state[s[0] = s[1]])
+            renderer.config.blockStack.push(
+                    <Link
+                        to={{
+                            pathname: "/search/text",
+                            state
+                        }}
+                        key={renderer.config.nextKey++}
+                        className="browserWord"
+                    >
+                        {data.payload}
+                    </Link>
+            );
+            } else {
+                renderer.config.blockStack.push(data.payload);
+            }
+        }
+    );
+    dInstance.addAction(
+        'scope',
+        () => true,
+        (renderer, context, data) => {
+            const scopeName = data.payload.split('/')[0];
+            const scopeValue = data.payload.split('/').slice(1).join('/');
+            if (data.subType === 'start' && scopeName === 'verses') {
+                renderer.config.currentVerses = scopeValue;
+                const currentChapter = renderer.config.currentChapter;
+                let verseRef = null;
+                let className = "verses_label";
+                if (
+                    renderer.config.currentChapter === renderer.config.selectedChapter &&
```

Comment:
with the current selection of gal 5 the code never enters this block, so am assuming that this has some future use